server/cdn-purging/flushPurgeQueue.ts

```diff
@@ -0,0 +1,23 @@
+import fastlyPurgeQueue from './fastlyPurgeQueue';
+import sendPurgeRequestToFastly from './sendPurgeRequestToFastly';
+
+export default function flushPurgeQueue() {
+  let keyBatch: string[] = [];
+  const purge = () => {
+    if (keyBatch.length) {
+      sendPurgeRequestToFastly(keyBatch);
+      keyBatch.forEach(key => fastlyPurgeQueue.remove(key));
+      keyBatch = [];
+    }
+  };
+  fastlyPurgeQueue
+    .find()
+    .forEach((doc) => {
```

Comment:
-1: This whole thing might actually be easier to read with a normal loop, and without purge as a local arrow function with a keyBatch binding. But this might just be my personal taste. Functionality is fine.

Reply:
The point here was that this is directly iterating the cursor without doing a full fetch, which means this can be run in batches with `getmore` directly from the mongodb cursor.

Reply:
Even if this looks like overkill for result sets this small – this could help in case that something doesn't work and we end up with a too big queue for some reason on a server start ;)

Reply:
I get the reason and the underlying idea is fine - I personally dislike unneeded bindings & nested functions - cursors have a `next()` and should support `for...of` as well. There is no need for a forEach or instantiating a different purge arrow function each call. But as said, this is mainly taste, I think the performance benefits of a less deep call stack do not matter here.